### PR TITLE
[stable/gocd] Deprecate agent.env.agentAutoRegisterEnvironemnts in favour of agent.env.agentAutoRegisterEnvironments (#9420)

### LINKED ISSUE #9420: [stable/gocd] typo in values: agent.env.agentAutoRegisterEnvironemnts
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
Bug report

**Version of Helm and Kubernetes**:
helm: 2.11.0, kubernetes 1.12.2

**Which chart**:
stable/gocd

**What happened**:
The value for setting enviroments for agents is named `agent.env.agentAutoRegisterEnvironemnts`, which seems to be a typo

**What you expected to happen**:
I would expect the value to be named `agent.env.agentAutoRegisterEnvironments`

**How to reproduce it** (as minimally and precisely as possible):
install the chart with this values file:
```
agent:
  env:
    agentAutoRegisterEnvironments: test 
```
the created agends won't have a environment set

**Anything else we need to know**:
Fixing this should be trivial and I have already fixed it in my fork, but technically fixing the typo would break backwarts-compatibility and I don't know how you want to handle this.

### PR DESCRIPTION
**Pull request by varshavaradarajan**
#### What this PR does / why we need it:
Deprecate agent.env.agentAutoRegisterEnvironemnts in favour of `agent.env.agentAutoRegisterEnvironments` due to a typo.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #9420 

#### Special notes for your reviewer:
Would be best if someone from the team could verify the change. Also, merge after #10397 so that @matthias-k need not have to fix merge conflicts. 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/gocd/values.yaml [hunk 1]
```diff
@@ -200,7 +200,10 @@ agent:
     # agent.env.agentAutoRegisterResources is the GoCD Agent auto-register resources
     agentAutoRegisterResources:
     # agent.env.agentAutoRegisterEnvironments is the GoCD Agent auto-register Environments
-    agentAutoRegisterEnvironemnts:
+    # deprecated because of a typo. Use agent.env.agentAutoRegisterEnvironments instead
+    agentAutoRegisterEnvironemnts: env1, env2
```
Comment (arvindsv) [inline stable/gocd/values.yaml]: By setting this here, won't it end up going through the `else if` clause?

Comment (varshavaradarajan) [reply stable/gocd/values.yaml]: Ah sorry, unintended checkin. Will fix it.

**Comment by arvindsv** [comment]
@ankitsri @sheroy: Can one you take a look at this at some point, maybe early next week? We need to verify a few scenarios:

1. The old value was not set (in `values.yaml`) and no new value is set. Everything continues to be fine on an upgrade.

2. The old value was set. New value is not set. Old value continues to be used.

3. The old value was not set. New value is set. New value is used.

4. The old value was set. New value is set. New value is used.

5. Removing the line with the old value from values.yaml doesn't make anything fail.

From the change, this should be true. But, would like to be sure.

**Comment by sheroy** [comment]
@arvindsv and @varshavaradarajan, I'm in the middle of testing this change. 

I set "agentAutoRegisterEnvironments: staging" in values.yaml. I expect to see an environment variable "AGENT_AUTO_REGISTER_ENVIRONMENTS" in the agent container's environment section once I describe an agent pod. I don't. What am I missing? 

/cc: @ankitsri11

**Comment by varshavaradarajan** [comment]
@sheroy - I don't know, sorry. Works for me on minikube. Are you on the right branch? Did doing helm repo add stable (you might have done this some time ago) interfere with the version of the chart that was installed? Did you check the version of the helm chart? Should be `1.5.14`. You can check that with `helm ls -a`.

My values.yaml
screen shot 2019-01-07 at 1 49 00 pm

Output of kubectl describe agent pod
screen shot 2019-01-07 at 1 52 49 pm

**Comment by sheroy** [comment]
@varshavaradarajan Never mind, I was looking at an elastic agent pod. Yep, the simple scenario works for me as well, just going through the other scenarios now.

**Comment by sheroy** [comment]
@varshavaradarajan and @arvindsv, I went through that list of scenarios, it all works fine. So lgtm.

**Pull request merged**